repl/src/metta_shim.rs

```diff
@@ -107,28 +131,96 @@ impl MettaShim {
             func(self)
         }}
     }
-}
 
-#[derive(Clone, PartialEq, Debug)]
-pub struct ImportPyErr;
+    pub fn get_config_atom(&mut self, config_name: &str) -> Option<Atom> {
+        let mut result = None;
+        metta_shim_env!{{
+            let repl_params = self.repl_params.borrow();
+            let config_metta_path = repl_params.config_metta_path();
+            let metta_modules = self.metta.modules().borrow();
+            let config_space = metta_modules.get(config_metta_path).unwrap();
+            let bindings_set = config_space.query(&Atom::expr(vec![sym!("="), Atom::sym(config_name.to_string()), expr!(val)]));
+            if let Some(bindings) = bindings_set.into_iter().next() {
+                result = bindings.resolve(&VariableAtom::new("val"));
+            }
```

Comment:
The following code should work (I didn't run it so please don't commit):
```suggestion
            let val = VariableAtom::new("val");
            let bindings_set = metta.space().query(&Atom::expr(vec![EQUAL_SYMBOL, Atom::sym(config_name.to_string()), Atom::Variable(val.clone())]));
            if let Some(bindings) = bindings_set.into_iter().next() {
                result = bindings.resolve(&val);
            }
```

Reply:
Yes.  That does work.  But that leads me to a different question... Is that what we want?

My thinking was that, by keeping the configs in separate space, it limited the number of potential name collisions with user-defined atoms.

I didn't realize the MeTTa default space ended up encompassing all the sub-spaces.

Reply:
The name collision exists since settings file is loaded as a subspace. This is how `import` works and changing this behavior is changing `import` semantics. On the other hand I don't see a big difference between importing settings and stdlib. I believe we also would like to allow user changing settings by changing the atomspace. I agree it would be better to have namespaces for the imported libraries and different modes of import: with importing all names into a global namespace and without. But it is not a question to this PR. Thus I would simplify the code here but anyway it is a minor thing.

Reply:
Some of the settings are loaded once - for efficiency (e.g. interpreter colors).  Others are reloaded each prompt.  (e.g. the prompt text itself, so you can do something like show some state associated with your space).

RE whether to query the global space or the subspace, you're absolutely right that, if we want to keep the config space merged, your simplification is better.

I was wondering if it might be be better to just load the configs into a space that isn't a proper subspace - but instead is just a space that needs to be accessed explicitly.

Reply:
> I was wondering if it might be be better to just load the configs into a space that isn't a proper subspace - but instead is just a space that needs to be accessed explicitly.

I am fine with either solution for this PR. In a golden state I would expect we will have ability to import settings with a separate namespace and allow user to edit it from a program.

Reply:
Honestly I am fine too.  It's less work to just load them into a subspace of the global space.

Since we're not quite sure what other behavior we want around accessing a partitioned space, I'll just load them into a subspace of the global space for now.

Is there some prefix we should give the symbols to make it less likely we'll encounter conflicts with user code?

Reply:
We could prefix them by `Repl` or something before we have namespaces for imported spaces.

Reply:
Ok.  Now all repl-specific config atoms begin with "Repl" in 327738f